[flang][OpenMP] Don't pass clauses to op-generating functions anymore

kparzysz opened the pull request:
Remove parameter `const List<Clause> &clauses` from functions that take construct queue.  The clauses should now be accessed from the construct queue.

tblah reviewed (approved):
LG. Thanks for the cleanup

skatrak reviewed (approved):
LGTM, thanks!

kparzysz committed:
[flang][OpenMP] Lower standalone ops via OMP dispatch, NFC

This moves lowering of standalone OpenMP ops into the dispatch
function. Follow-up to PR90098.

kparzysz committed:
aa

kparzysz committed:
clang-format, alphabetize directives in switch statement

kparzysz committed:
Differentiate standalone and block-associated variants of "ordered"

kparzysz committed:
Add "section" to dispatch as well

kparzysz committed:
Rebase on top of new changes

kparzysz committed:
Merge branch 'main' into users/kparzysz/spr/d10-noclauses

kparzysz committed:
Fix merge errors
